view/src/service.rs

```diff
@@ -513,14 +513,12 @@ impl ViewProtocolService for ViewService {
                     "Error retrieving transaction by hash {}",
                     hex::encode(&request.tx_hash)
                 ))
-            })?
-            .ok_or_else(|| {
-                tonic::Status::failed_precondition(format!(
-                    "No transaction found with this hash {}",
-                    hex::encode(&request.tx_hash)
-                ))
             })?;
 
+        let Some(tx) = maybe_tx else {
+            return Ok(tonic::Response::new(pb::TransactionPerspectiveResponse::default()));
+        };
```

Comment:
Not totally happy with this -- what if the view service didn't have the transaction, but it was still viewable in principle?  (i.e., what if we didn't have the current behavior where pcli always downloads every matching transaction before continuing its sync process?)

Reply:
Maybe we want something like a, uh, phantom response? Or pending? That way we can indicate this state of affairs to the user at least?

Reply:
I think the bigger issue is: currently we make sync speed slower in order to have the property that, if the view service reports it has synced to height `h`, it's guaranteed to have _all_ relevant data up to height `h`.

So, for instance, as soon as we report that we've synced a block, we're guaranteed to have the full transaction info for anything we might have detected.

However, basically every trick that we could do to make sync faster / lighter than it is now would require breaking that assumption.

For instance, we probably don't want to stop scanning once we detect something (because we're waiting to fetch full details in the background), but that means that if we notify someone about a detection, they might not be able to immediately download the relevant transaction.  Is that a problem?

Or, we could do "DAGSync", where as soon as we detect one transaction, we query for transactions spending its output notes, and walk the DAG of our own transaction graph.  This means we'll end up with a "partial" view of sync state up to the chain tip, which we progressively backfill in the background.  What should the query behavior be in that case?

As long as we have the current behavior, we don't really have to worry about it, so the current code in this specific place is currently fine -- but it points to this bigger issue of how we want to handle partially-synced state.

Reply:
we could add an additional function mode? something like "full sync" vs. "fast sync" where fast sync will return some phantom/pending tx info as you describe that returns a partial view of the sync state. Could we make that as an additive change? Because the existing behavior seems worth keeping as an option even if it's more time-consuming